packages/core/core-flows/src/order/steps/add-order-transaction.ts

```diff
@@ -5,48 +5,34 @@ import { StepResponse, createStep } from "@medusajs/framework/workflows-sdk"
 /**
  * The transaction(s) to add to the order.
  */
-export type AddOrderTransactionStepInput = CreateOrderTransactionDTO | CreateOrderTransactionDTO[]
+export type AddOrderTransactionStepInput =
+  | CreateOrderTransactionDTO
+  | CreateOrderTransactionDTO[]
 
 /**
  * The added order transaction(s).
  */
-export type AddOrderTransactionStepOutput = CreateOrderTransactionDTO | CreateOrderTransactionDTO[]
+export type AddOrderTransactionStepOutput =
+  | CreateOrderTransactionDTO
+  | CreateOrderTransactionDTO[]
 
 export const addOrderTransactionStepId = "add-order-transaction"
 /**
  * This step creates order transactions.
  */
 export const addOrderTransactionStep = createStep(
   addOrderTransactionStepId,
-  async (
-    data: AddOrderTransactionStepInput,
-    { container }
-  ) => {
+  async (data: AddOrderTransactionStepInput, { container }) => {
     const service = container.resolve(Modules.ORDER)
 
     const trxsData = Array.isArray(data) ? data : [data]
 
-    for (const trx of trxsData) {
-      const existing = await service.listOrderTransactions(
-        {
-          order_id: trx.order_id,
-          reference: trx.reference,
-          reference_id: trx.reference_id,
-        },
-        {
-          select: ["id"],
-        }
-      )
-
-      if (existing.length) {
-        return new StepResponse(null)
```

Comment:
Hey @carlos-r-l-rodrigues , can you think of any reason why this would skip if there is an existing transaction?

Reply:
The reference_id here should not repeat. Every transaction ideally has it's own reference_id.
A different return_id, a different payment_id, or capture_id, anything.
but ideally not duplicated.

Reply:
This is on purpose to avoid duplicated transactions

Reply:
@riqwan, @carlos-r-l-rodrigues did we get to a resolution on this?